File: Counting Sort/counting_sort.cs
internal static int MaximumNumber(int[] arr)
{
    if (arr.Length == 0) return 0;
    int max = arr[0];
    for (int i = 1; i < arr.Length; ++i)
        if (max < arr[i])
            max = arr[i];
    return max;
}
internal static int[] CountingSort(int[] arr)
{
    int temp = MaximumNumber(arr);
    int n = arr.Length > temp ? arr.Length : temp + 1;
    int[] tempArr = new int[n];
    for (int i = 0; i < arr.Length; i++) tempArr[arr[i]]++;
    int j = 0;
    for (int i = 0; i < n; i++)
        while (tempArr[i]-- != 0)
            arr[j++] = i;
    return arr;
}

File: Counting Sort/counting_sort.py
def MaximumNumber(arr):
    if len(arr) == 0: return 0
    mx = arr[0]
    for i in range(1, len(arr)):
        if mx < arr[i]:
            mx = arr[i]
    return mx

def CountingSort(arr):
    tempNum = MaximumNumber(arr)
    n = len(arr) if len(arr) > tempNum else tempNum + 1
    tempArr = [0] * n
    for i in range(len(arr)): tempArr[arr[i]]+= 1
    j = 0
    for i in range(n):
        while tempArr[i] != 0:
            arr[j] = i
            tempArr[i]-= 1
            j += 1
    return arr